server/controller/store.go

```diff
@@ -29,16 +31,23 @@ var storeLogger = logging.Logger("controller/store")
 
 type storeCtrl struct {
 	storev1.UnimplementedStoreServiceServer
-	store types.StoreAPI
-	db    types.DatabaseAPI
+	store     types.StoreAPI
+	db        types.DatabaseAPI
+	validator *service.ValidationService
 }
 
-func NewStoreController(store types.StoreAPI, db types.DatabaseAPI) storev1.StoreServiceServer {
+func NewStoreController(store types.StoreAPI, db types.DatabaseAPI) (storev1.StoreServiceServer, error) {
```

Comment:
Can we create a validation interface and add validation dependency as we do with Store and Database?